lib/msal-common/src/client/RefreshTokenClient.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Configuration } from "../config/Configuration";
+import {BaseClient} from "./BaseClient";
+import {RefreshTokenRequest} from "../request/RefreshTokenRequest";
+import {Authority, NetworkResponse} from "..";
+import {ServerAuthorizationTokenResponse} from "../server/ServerAuthorizationTokenResponse";
+import {RequestParameterBuilder} from "../server/RequestParameterBuilder";
+import {ScopeSet} from "../request/ScopeSet";
+import {GrantType} from "../utils/Constants";
+
+/**
+ * OAuth2.0 refresh token client
+ */
+export class RefreshTokenClient extends BaseClient {
+
+    constructor(configuration: Configuration) {
+        super(configuration);
+    }
+
+    public async acquireToken(request: RefreshTokenRequest): Promise<string>{
+        const authority = await this.createAuthority(request.authority);
+        const response = await this.executeTokenRequest(request, authority);
+        // TODO add response_handler here to send the response
+        return JSON.stringify(response.body);
+    }
+
+    private async executeTokenRequest(request: RefreshTokenRequest, authority: Authority)
+        : Promise<NetworkResponse<ServerAuthorizationTokenResponse>> {
```

Comment:
Update: Can we have the mapping changes added with this PR?

Reply:
@sameerag What mapping changes are you referring to?

Reply:
I think it is ok to have those changes in a separate PR.